src/java.desktop/macosx/native/libosxapp/JNIUtilities.h

```diff
@@ -29,11 +29,24 @@
 #include "jni.h"
 #include "jni_util.h"
 
+#import <Cocoa/Cocoa.h>
+
 /********        LOGGING SUPPORT    *********/
 
 #define LOG_NULL(dst_var, name) \
    if (dst_var == NULL) { \
        NSLog(@"Bad JNI lookup %s\n", name); \
+       NSLog(@"%@",[NSThread callStackSymbols]); \
+       if ([NSThread isMainThread] == NO) { \
+           if ((*env)->ExceptionOccurred(env) == NULL) { \
```

Comment:
> Not sure that the check for ExceptionOccurred is needed, 
It may not be needed in practice but if the code path is never taken no harm ...

in all other places where we check the ref to methods/field we only check the return value, and if it is null then return immediately assuming that an exception is rased already, for example :
> 
> https://github.com/openjdk/jdk/blob/b72de3c5fc99f365e9fb25114ddd28eceddfa6e8/src/java.desktop/windows/native/libawt/windows/awt_Button.cpp#L357
> 
> Note that the exception in the static initializer is fatal for the application.

Nothing new here.

Reply:
The new thing here is that you check the result of the ExceptionOccurred if NULL is returned from the GetMethodID/etc, we usually never do it. If such checks are necessary then I'll create a separate bug to update other similar use cases.

Reply:
Here I don't have immediate control over why this is being called.
So we could have null but no pending exception depending on what the caller did.
It won't be executed except in the rare case there's a problem so it certainly isn't causing overhead so what is the problem ?
I don't see a need to change other code unless there's a really good reason.